---
title: "Manuscript_4_Molecular_Analysis_directory_RBRA"
author: "Pieter van Veelen"
date: "5/1/2021"
output: html_document
---

This is an R Markdown document contains an analysis of microbial community dynamics of an experiment of Rita H.R. Branco in collaboration with H. Pieter J. van Veelen.

## Install packages

```{r setup, include=T}
knitr::opts_knit$set(root.dir = "C:/Users/Rita/Documents/R/amendment_micropollutant_biodegradation_aquifer")

knitr::opts_chunk$set(echo = TRUE)

options(scipen = 999)

# packages required
packages <- c("qiime2R", "phyloseq", "tidyverse", "magrittr", "devtools", "qiime2R", "here", "scales", "breakaway", "DivNet", "openxlsx", "ape", "vegan", "ggtext", "cowplot", "RColorBrewer", "microbiome", "lme4", "lmerTest","decontam", "remotes", "ampvis2", "speedyseq", "ggh4x", "kableExtra", "phia", "MASS", "ggord", "glue", "lubridate", "data.table", "ggplot2")

# install packages not yet installed
installed_packages <- packages %in% rownames(installed.packages())
if (any(installed_packages == FALSE)) {
  install.packages(packages[!installed_packages])
}

# # packages not available for R 4.1.2
# ## phyloseq
# if(!requireNamespace("BiocManager")){install.packages("BiocManager")}
# BiocManager::install("phyloseq")
# 
# ## qiime2R
# if (!requireNamespace("devtools", quietly = TRUE)){install.packages("devtools")}
# devtools::install_github("jbisanz/qiime2R")
# 
# ## breakaway
# install.packages("devtools")
# devtools::install_github("adw96/breakaway")
# 
# ## DivNet
# remotes::install_github("adw96/breakaway")
# remotes::install_github("adw96/DivNet")
# 
# ## microbiome
# library(BiocManager)
# BiocManager::install("microbiome")
# 
# ## decontam
# if (!requireNamespace("BiocManager", quietly = TRUE))
# install.packages("BiocManager")
# BiocManager::install("decontam")
# 
# ## ampvis2
# if (!requireNamespace("ampvis2", quietly = TRUE))
# install.packages("ampvis2")
# remotes::install_github("MadsAlbertsen/ampvis2", Ncpus = 6)
# 
# ## speedyseq
# if (!requireNamespace("speedyseq", quietly = TRUE))
#     install.packages("speedyseq")
# remotes::install_github("mikemc/speedyseq")
# 
# ## ggord
# if (!requireNamespace("ggord", quietly = TRUE))
# install.packages("remotes")
# remotes::install_github("fawda123/ggord")
```

## Exploratory data analysis

### Loading libraries

```{r library loading, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# load required packages
library(qiime2R)
library(phyloseq)
library(tidyverse)
library(magrittr)
library(devtools)
library(qiime2R)
library(here)
library(breakaway)
library(DivNet)
library(openxlsx)
library(ape)
library(vegan)
library(ggtext)
library(cowplot)
library(RColorBrewer)
library(microbiome)
library(lme4)
library(lmerTest)
library(decontam)
library(remotes)
library(ampvis2)
library(speedyseq)
library(ggh4x)
library(ggord)
library(glue)
library(lubridate)
library(phia)
library(kableExtra)
library(MASS)
library(data.table)
library(scales)
library(ggplot2)
```

### Data import

Below, the input data are imported in R. These files are output files from our QIIME2 (v.2019.10) sequence data pre-processing workflow. Scripts and parameter settings are found in separate bash files.

```{r import data, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# creating phyloseq objects with 
physeq = qza_to_phyloseq(features = "input_data/RBRA_16S_515F926R_Q11696_20230330_table.qza",
                        tree = "input_data/RBRA_16S_515F926R_Q11696_20230330_rooted-tree.qza",
                        taxonomy = "input_data/NB_classifier_SILVA_138_99_16S_515F-926R_QIIME2-2022.11.qza",
                        metadata = "input_data/RBRA_16S_515F926R_Q11696_20230330@metadata.txt")
```

### Cleaning data set

The following steps are subsequently performed to clean the data: 1) bifurcating tree using ape package; 2) cleaning up the metadata; 3) replacing taxonomic strings that are empty, NA, metagenome, ambiguous taxa; 4) decontaminating based on frequency and prevalence using the DECONTAM package; 5) split blanks from samples; 

#### 1) Resolve phylogenetic tree

Number of tips = number internal nodes + 1. If this does not happens, then nodes from the phylogenetic tree have to be splited/bifurcated.

```{r clean phylogeny, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# evaluate tree topology
is.binary(phy_tree(physeq))

# if FALSE -> resolve polychotomous nodes
phy_tree_resolved <- multi2di(phy_tree(physeq))
is.binary(phy_tree_resolved)

# create new phy_tree
tree2 <- phy_tree_resolved

# merge new phy_tree object with sample_data and otu_table into new phyloseq object
psdata_RBRA <- merge_phyloseq(otu_table(physeq), sample_data(physeq), tax_table(physeq), tree2)

# confirmation polychotomous nodes resolved
is.binary(phy_tree(psdata_RBRA))
```

#### 2) Clean-up metadata

```{r clean metadata, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
sample_data(psdata_RBRA)$column <- factor(sample_data(psdata_RBRA)$column, levels = c("Nitrate_red_Microaer", "Nitrate_red", "Aerob", "Aerob_Microaer"))

sample_data(psdata_RBRA)$e_donor <- factor(sample_data(psdata_RBRA)$e_donor, levels = c("none", "H", "H_D", "D", "D_A", "A", "NH4"))

sample_data(psdata_RBRA)$height <- factor(sample_data(psdata_RBRA)$height, levels = c("2.5", "50"))

sample_data(psdata_RBRA)$day_DOC <- factor(sample_data(psdata_RBRA)$day_DOC, levels = c("H_14", "H_D_42",	"D_70",	"D_A_98",	"A_126",	"NH4_154",	"Nitrate_red_14",	"Microaerob_70", "Microaerob_126",	"Microaerob_154"))

sample_data(psdata_RBRA)$column_height <- factor(sample_data(psdata_RBRA)$column_height, levels = c("Nitrate_red_Microaer_50",	"Nitrate_red_2.5", "Aerob_2.5", "Aerob_Microaer_2.5", "Aerob_Microaer_50"))


names(sample_data(psdata_RBRA)) <- tolower(names(sample_data(psdata_RBRA)))
```

#### 3) Clean-up taxa data

```{r clean taxanomy, message=F, echo=T, eval=T, warning=F, include=T, cache=F}
# remove Chloroplast and Mitochondia
physeq1 <- subset_taxa(psdata_RBRA, Order != "Chloroplast")
physeq2 <- subset_taxa(physeq1, Family != "Mitochondria")

# clean taxonomy tags with no information, specify NA taxon name tags to last known taxon names
tax.clean1 <- data.frame(tax_table(physeq2))

tax.clean2 <- tax.clean1
tax.clean2$Kingdom <- as.character(tax.clean2$Kingdom)
tax.clean2$Phylum <- as.character(tax.clean2$Phylum)
tax.clean2$Class <- as.character(tax.clean2$Class)
tax.clean2$Order <- as.character(tax.clean2$Order)
tax.clean2$Family <- as.character(tax.clean2$Family)
tax.clean2$Genus <- as.character(tax.clean2$Genus)
tax.clean2$Species <- as.character(tax.clean2$Species)

# remove ambiguous taxa
tax.clean = tax.clean2 %>% mutate_all(funs(str_replace(., ".*metagenome.*", "")))
tax.clean = tax.clean %>% mutate_all(funs(str_replace(., ".*uncultured.*", "")))
tax.clean = tax.clean %>% mutate_all(funs(str_replace(., ".*unidentified.*", "")))
tax.clean = tax.clean %>% mutate_all(funs(str_replace(., ".*groundwater.*", "")))
tax.clean = tax.clean %>% mutate_all(funs(str_replace(., "bacterium_enrichment", "")))
tax.clean = tax.clean %>% mutate_all(funs(str_replace(., "denitrifying_bacterium", "")))
tax.clean = tax.clean %>% mutate_all(funs(str_replace(., "agricultural_soil", "")))
tax.clean[is.na(tax.clean)] <- ""

# function replace name tags from [https://github.com/joey711/phyloseq/issues/850]
for (i in 1:nrow(tax.clean)){
  if (tax.clean[i,2] == ""){
    Kingdom <- paste("Kingdom_", tax.clean[i,1], sep = "")
    tax.clean[i, 2:7] <- Kingdom
  } else if (tax.clean[i,3] == ""){
    Phylum <- paste("Phylum_", tax.clean[i,2], sep = "")
    tax.clean[i, 3:7] <- Phylum
  } else if (tax.clean[i,4] == ""){
    Class <- paste("Class_", tax.clean[i,3], sep = "")
    tax.clean[i, 4:7] <- Class
  } else if (tax.clean[i,5] == ""){
    Order <- paste("Order_", tax.clean[i,4], sep = "")
    tax.clean[i, 5:7] <- Order
  } else if (tax.clean[i,6] == ""){
    Family <- paste("Family_", tax.clean[i,5], sep = "")
    tax.clean[i, 6:7] <- Family
  } else if (tax.clean[i,7] == ""){
    tax.clean$Species[i] <- paste("Genus_",tax.clean$Genus[i], sep = "_")
  }
}

# save old tax table
psdata_oldTaxtab <- psdata_RBRA

# put cleaned tax_table into phyloseq object
tax_table(physeq2) <- as.matrix(tax.clean)

psdata_RBRA <- physeq2
```

#### 4) Remove data from experiment 3

```{r remove data exp 3, message=F, echo=T, eval=T, warning=F, include=T, cache=F}
# save old phyloseq
psdata_RBRA_complete <- psdata_RBRA

physeq3 <- subset_samples(psdata_RBRA, experiment == 4)

physeq3 <- prune_taxa(taxa_sums(physeq3)>0, physeq3)
```

#### 5) Remove OTUs with origin in MOCK community
```{r remove MOCK OTUs, message=F, echo=T, eval=T, warning=F, include=T, cache=F}
mock <- subset_samples(physeq3, mock_community == "yes")

rel_not_rar <- transform_sample_counts(physeq3, fun = function(x) x/sum(x)) ## whole data set
rel_mock <- transform_sample_counts(mock, fun = function(x) x/sum(x)) ## mock community

# taxa in MOCK samples with abundance > 0.01
filter_0.01 <- phyloseq::genefilter_sample(rel_mock, filterfun_sample(function(x) x > 0.01))
mock_taxa_0.01 <- prune_taxa(filter_0.01, rel_mock)
not_rar_0.01 <- prune_taxa(filter_0.01,rel_not_rar) #11 OTU, all these taxa correspond to taxa that should be in the mock community, their relative abundance in the samples (0.000-0.003) is much lower than in the mock community samples (>0.010)

# taxa in MOCK samples with abundance > 0.005
filter_0.005 <- phyloseq::genefilter_sample(rel_mock, filterfun_sample(function(x) x > 0.005))
mock_taxa_0.005 <- prune_taxa(filter_0.005, rel_mock) 
not_rar_0.005 <- prune_taxa(filter_0.005, rel_not_rar) #15 OTU

# taxas in MOCK samples with 0.005 < abundance > 0.01
filter_0.005_0.01 <- phyloseq::genefilter_sample(mock_taxa_0.005, filterfun_sample(function(x) x < 0.01))
mock_taxa_0.005_0.01 <- prune_taxa(filter_0.005_0.01, mock_taxa_0.005) 
not_rar_0.005_0.01 <- prune_taxa(filter_0.005_0.01,not_rar_0.005) #4 OTU
### 3eff8f2d520a354e57817f9210dc426d (Genus__Lysinibacillus, mock taxa) mock: 0.0042-0.0055; sample: 0.0000 -> from MOCK sample
### 599b022f8ad5fc64b1423b0931dd052e (Bacillus_coagulans, mock taxa) mock: 0.0044-0.0100; sample: 0.0000 -> from MOCK sample
### 500d294e65ac031e205edb2ea3b9ddf6 (Family_Comamonadaceae, not mock taxa) mock: 0.0043-0.0073; sample: up to 0.21 -> experimental sample contaminated MOCK sample
### eee4b43e99303cb0092fe501da59d116 (Genus__Clostridium_sensu_stricto_1, not mock taxa) mock: 0.0077-0.0087; sample: up to 0.42 -> experimental sample contaminated MOCK sample

# filter out of experiment samples the taxa from the MOCK samples
not_mock_taxa_1 <- prune_taxa(!filter_0.01, physeq3)
other_mock_taxa = c("3eff8f2d520a354e57817f9210dc426d",  "599b022f8ad5fc64b1423b0931dd052e")
not_mock_taxa_2 <- setdiff(taxa_names(not_mock_taxa_1), other_mock_taxa)
physeq4 <- prune_taxa(not_mock_taxa_2, not_mock_taxa_1)

# remove MOCK samples
psdata_RBRA <- subset_samples(physeq4, mock_community == "no")
```

#### 6) Remove contaminants with DECONTAM

```{r decontam, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# add sample_or_control column
sample_data(psdata_RBRA)$is.neg <- sample_data(psdata_RBRA)$negative_control == "yes"
sample_data(psdata_RBRA)$sample_or_control <- factor(if_else(sample_data(psdata_RBRA)$negative_control == "yes", "control", "sample"))

# replace DNA concentration = 0 for a very low DNA amount
sample_data(psdata_RBRA)$dna_conc <- if_else(sample_data(psdata_RBRA)$dna_conc == 0, sample_data(psdata_RBRA)$dna_conc+0.0001, sample_data(psdata_RBRA)$dna_conc)

df <- as.data.frame(sample_data(psdata_RBRA)) # put sample_data into a ggplot-friendly data.frame
df$librarysize <- sample_sums(psdata_RBRA) # 'samples_sums' returns the total number of individuals observed from each sample
df <- df[order(df$librarysize),] # 'order' returns a permutation which rearranges its first argument into ascending or descending order
df$index <- seq(nrow(df)) # 'seq' generates regular sequences, 'nrow' return the number of rows or columns present in an array-like object

df %>%    
ggplot(aes(x=index, y=librarysize, color=sample_or_control)) + 
  geom_point() + 
  scale_y_continuous(trans = "log10") +
  labs(y = "Library size (log10)", x = "Sample rank") +
  theme_classic()

# contaminants based on prevalence due to low DNA concentration
contamdf.prev <- isContaminant(psdata_RBRA, method="prevalence", neg="is.neg", threshold=0.1)
n_contam_prev <- table(contamdf.prev$contaminant) # 13 contaminants based on prevalence

# Make phyloseq object of presence-absence in negative controls and true samples
ps.pa <- transform_sample_counts(psdata_RBRA, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$sample_or_control == "control", ps.pa) 
ps.pa.pos <- prune_samples(sample_data(ps.pa)$sample_or_control == "sample", ps.pa)

# Make dataframe of prevalence in positive and negative samples
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                      contaminant=contamdf.prev$contaminant)
ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() +
  xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)")

# saving contaminant and decontaminated data
psdata_contam <- prune_taxa(contamdf.prev$contaminant, psdata_RBRA)
psdata_decontam <- prune_taxa(!contamdf.prev$contaminant, psdata_RBRA)
```

#### 7) Remove blanks from the dataset

```{r select samples, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# add SampleID
sample_data(psdata_decontam)$sampleid <- sample_names(psdata_decontam)

# keep the taxa that are present on one ore more of the samples
psdata_RBRA <- 
  prune_taxa(
    taxa_sums(
      subset_samples(psdata_decontam, negative_control == "no")) > 0, 
      subset_samples(psdata_decontam, negative_control == "no"))

psdata_RBRA <- subset_samples(psdata_RBRA, !column_height == "Aerob_Microaer_50")

psdata_RBRA <- prune_taxa(taxa_sums(psdata_RBRA)>0, psdata_RBRA)
```

### Rarefaction curves

#### Filter on abundance 

```{r filter on abundance, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# relative abundance data
psdata_RBRA_rel <- transform_sample_counts(psdata_RBRA, fun = function(x) x/sum(x)) # 1657 taxa

# total reads left = 1532725
sum(sample_sums(psdata_RBRA))

# continue downstream analysis with abundance filter that retains ASVs with at least 0.01% of total read abundance -> keeps above 99% abundance when removing rare data
psdata_RBRA_0.01pct <- prune_taxa(taxa_sums(psdata_RBRA_rel) > 0.0001, psdata_RBRA) # 1334 taxa (99.93% abundance)
100*(sum(sample_sums(psdata_RBRA_0.01pct))/sum(sample_sums(psdata_RBRA)))

psdata_RBRA <- psdata_RBRA_0.01pct
```

#### Reshape metadata

```{r reshape metadata, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
metadata = 
  data.frame(sample_data(psdata_RBRA), stringsAsFactors = T) %>% as_tibble() %>% 
  rename_all(. %>% tolower) %>% 
  mutate(combined = factor(paste0(day,"-",column,"-",height,"-",e_donor)))

meta_formatted = sample_data(metadata)
sample_names(meta_formatted) = metadata$description
sample_data(psdata_RBRA) = meta_formatted
```

#### Plotting rarefaction curves

```{r rarefaction curves, message=F, echo=T, eval=T, warning=F, include=T, cache=F, fig.width=4.5, fig.height=3}
# data frame in which the rows are OTU IDs, the columns are samples, and the last 7 columns are the corresponding taxonomy assigned to the OTUs
otus_tax = cbind(
  as.data.frame(psdata_RBRA@otu_table),
  as.data.frame(as.matrix(psdata_RBRA@tax_table)))

# data frame with metadata
met = data.frame(psdata_RBRA@sam_data)
met = dplyr::select(met, sampleid, everything())

# load data
data <- amp_load(otutable = otus_tax,
                 metadata = met)

# minimum coverage of reads per sample
min_cover <- min(sample_sums(psdata_RBRA)) #26444
max_cover <- max(sample_sums(psdata_RBRA))
median_cover <- mean(sample_sums(psdata_RBRA))
sum_cover <- sum(sample_sums(psdata_RBRA))

# rarefaction curves plot
breakscolumn <- c("Nitrate_red_Microaer_50", "Nitrate_red_2.5", "Aerob_2.5", "Aerob_Microaer_2.5")
valuesfill <- c("#CA0020", "#F4A582", "#92C5DE", "#0571B0")

amp_rarecurve(data = data, 
              color = "column_height") +
  scale_color_manual (breaks = breakscolumn,
                      values = valuesfill) +
  theme_classic() +
  theme(axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.position ="bottom",
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5)) +
  coord_cartesian(xlim = c(0,50000), ylim = c(0,500)) +
  geom_vline(xintercept = 26444, size = 0.25)

ggsave("C:/Users/Rita/Documents/R/amendment_micropollutant_biodegradation_aquifer/figures/Figure4_S1.pdf", width = 4.5, height = 3)
```

**Figure S1**. Rarefaction curves. Minimum coverage was 26444 reads per sample; a depth at which full diversity is captured. Hence, no samples were dropped in later stages.<br>

### Alpha diversity

```{r alpha diversity, message=F, echo=T, eval=T, warning=T, include=T, cache=T}
# data rarefied to 26445 r/s
set.seed(711)

ps_26444 <- rarefy_even_depth(psdata_RBRA, 26444, rngseed = 711)

# calculate richness
metadata <- data.frame(sample_data(ps_26444))
alpha <- estimate_richness(ps_26444, measures = c("Chao1", "Shannon"))

alpha$sampleid <- sample_names(ps_26444)
alpha_data <- 
  left_join(metadata, alpha, by = "sampleid") %>% 
  rename_all(tolower) 

alpha_summary = alpha_data %>% group_by(combined) %>% summarise(mean_chao = mean(chao1), n = n(), mean_shannon = mean(shannon), n = n())

# plot alpha populations bars
chao1 <- alpha_data %>% 
  ggplot(aes(x=day_doc, y=chao1)) +
  stat_summary(fun = mean, geom = "bar", mapping = aes(fill = column_height)) +
  scale_fill_manual(values = c("#CA0020", "#F4A582", "#92C5DE", "#0571B0")) +
  facet_grid(cols = vars(column_height), scales = "free_x", space = "free_x") +
  labs(y = "ASV richness (Chao1)", x= "Day_DOC") +
  scale_y_continuous(limits = c(-5, 505), 
                     breaks = seq(0, 500, 100), 
                     expand = c(0,0)) +
  theme_classic() +
  theme(axis.text = element_text(size = 7, colour = "black"),
        axis.ticks.x = element_line(size = 0.35),
        axis.ticks.y = element_line(size = 0.35),
        axis.text.x	= element_text(angle = 45, hjust = 1),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.position = "none",
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))

shannon <- alpha_data %>% 
  ggplot(aes(x=day_doc, y=shannon)) +
  stat_summary(fun = mean, geom = "bar", mapping = aes(fill = column_height)) +
  scale_fill_manual(values = c("#CA0020", "#F4A582", "#92C5DE", "#0571B0")) +
  facet_grid(cols = vars(column_height), scales = "free_x", space = "free_x") +
  labs(y = "ASV diversity (Shannon)", x= "Day_DOC") +
  scale_y_continuous(limits = c(-0.05, 5.05), 
                     breaks = seq(0, 5, 1), 
                     expand = c(0,0),
                     labels = label_number(accuracy = 0.01)) +
  theme_classic() +
  theme(axis.text = element_text(size = 7, colour = "black"),
        axis.ticks.x = element_line(size = 0.35),
        axis.ticks.y = element_line(size = 0.35),
        axis.text.x	= element_text(angle = 45, hjust = 1),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.position = "none",
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))

alpha_plot = plot_grid(chao1 + theme(legend.position = "none"), 
                       shannon + theme(legend.position = "none"),
                       align = "vh",
                       labels = c("A", "B"),
                       ncol = 2)
```

### Beta diversity

Due to no replicates no PERMANOVA was performed.

#### In columns with electron donor amendment

```{r beta diversity electron donor amendment, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# relative abundance data
psdata_RBRA_rel <- transform_sample_counts(psdata_RBRA, fun = function(x) x/sum(x))

# remove column no electon donor amendment
psdata_RBRA_rel_donor <- subset_samples(psdata_RBRA_rel, !e_donor == "none")

# ordination
PCoA_BC_donor <- ordinate(psdata_RBRA_rel_donor, method = "PCoA", distance = "bray")
PCoA_Jac_donor <- ordinate(psdata_RBRA_rel_donor, method = "PCoA", distance = "jaccard")
PCoA_uu_donor <- ordinate(psdata_RBRA_rel_donor, method = "PCoA", distance = "uunifrac")
PCoA_wu_donor <- ordinate(psdata_RBRA_rel_donor, method = "PCoA", distance = "wunifrac")

# Plot PCoA Bray-Curtis dissimilarity distances
plot_PCoA_BC_donor <- plot_ordination(physeq = psdata_RBRA_rel_donor, 
                                 ordination = PCoA_BC_donor, 
                                 type = "samples", 
                                 axes = c(1,2), 
                                 shape = "day_doc",
                                 color = "column_height") +
  scale_color_manual(name = "Column_height",
                     values = c("#F4A582", "#92C5DE", "#0571B0")) +
  scale_shape_manual(name = "Day_DOC",
                       values = c(0,6,1,2,5,4)) +
  geom_point(size = 2) +
  ggtitle("Bray-Curtis dissimilarity") +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))

# plot PCoA Jaccard distances
plot_PCoA_Jac_donor <- plot_ordination(physeq = psdata_RBRA_rel_donor, 
                                 ordination = PCoA_Jac_donor, 
                                 type = "samples", 
                                 axes = c(1,2), 
                                 shape = "day_doc",
                                 color = "column_height") +
  scale_color_manual(name = "Column_height",
                     values = c("#F4A582", "#92C5DE", "#0571B0")) +
  scale_shape_manual(name = "Day_DOC",
                       values = c(0,6,1,2,5,4)) +
  geom_point(size = 2) +
  ggtitle("Jaccard distance") +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))

# plot PCoA unweighted UniFrac distances
plot_PCoA_uu_donor <- plot_ordination(physeq = psdata_RBRA_rel_donor, 
                                 ordination = PCoA_uu_donor, 
                                 type = "samples", 
                                 axes = c(1,2), 
                                 shape = "day_doc",
                                 color = "column_height") +
  scale_color_manual(name = "Column_height",
                     values = c("#F4A582", "#92C5DE", "#0571B0")) +
  scale_shape_manual(name = "Day_DOC",
                       values = c(0,6,1,2,5,4)) +
  geom_point(size = 2) +
  ggtitle("unweighted UniFrac") +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))

# plot PCoA weighted UniFrac distances
plot_PCoA_wu_donor <- plot_ordination(physeq = psdata_RBRA_rel_donor, 
                                 ordination = PCoA_wu_donor, 
                                 type = "samples", 
                                 axes = c(1,2), 
                                 shape = "day_doc",
                                 color = "column_height") +
  scale_color_manual(name = "Column_height",
                     values = c("#F4A582", "#92C5DE", "#0571B0")) +
  scale_shape_manual(name = "Day_DOC",
                       values = c(0,6,1,2,5,4)) +
  geom_point(size = 2) +
  ggtitle("weighted UniFrac") +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))
```

#### In column without electron donor amendment

```{r beta diversity no electron donor amendment, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# remove column electon donor amendment
psdata_RBRA_rel_microaerob <- subset_samples(psdata_RBRA_rel, e_donor == "none")

# ordination
PCoA_BC_microaerob <- ordinate(psdata_RBRA_rel_microaerob, method = "PCoA", distance = "bray")
PCoA_Jac_microaerob <- ordinate(psdata_RBRA_rel_microaerob, method = "PCoA", distance = "jaccard")
PCoA_uu_microaerob <- ordinate(psdata_RBRA_rel_microaerob, method = "PCoA", distance = "uunifrac")
PCoA_wu_microaerob <- ordinate(psdata_RBRA_rel_microaerob, method = "PCoA", distance = "wunifrac")

# Plot PCoA Bray-Curtis dissimilarity distances
plot_PCoA_BC_microaerob <- plot_ordination(physeq = psdata_RBRA_rel_microaerob, 
                                 ordination = PCoA_BC_microaerob, 
                                 type = "samples", 
                                 axes = c(1,2), 
                                 shape = "day_doc",
                                 color = "column_height") +
  scale_color_manual(name = "Column_height",
                     values = c("#CA0020")) +
  scale_shape_manual(name = "Day_DOC",
                       values = c(0,1,5,4)) +
  geom_point(size = 2) +
  ggtitle("Bray-Curtis dissimilarity") +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))
# plot PCoA Jaccard distances
plot_PCoA_Jac_microaerob <- plot_ordination(physeq = psdata_RBRA_rel_microaerob, 
                                 ordination = PCoA_Jac_microaerob, 
                                 type = "samples", 
                                 axes = c(1,2), 
                                 shape = "day_doc",
                                 color = "column_height") +
  scale_color_manual(name = "Column_height",
                     values = c("#CA0020")) +
  scale_shape_manual(name = "Day_DOC",
                       values = c(0,1,5,4)) +
  geom_point(size = 2) +
  ggtitle("Jaccard distance") +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))

# plot PCoA unweighted UniFrac distances
plot_PCoA_uu_microaerob <- plot_ordination(physeq = psdata_RBRA_rel_microaerob, 
                                 ordination = PCoA_uu_microaerob, 
                                 type = "samples", 
                                 axes = c(1,2), 
                                 shape = "day_doc",
                                 color = "column_height") +
  scale_color_manual(name = "Column_height",
                     values = c("#CA0020")) +
  scale_shape_manual(name = "Day_DOC",
                       values = c(0,1,5,4)) +
  geom_point(size = 2) +
  ggtitle("unweighted UniFrac") +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))

# plot PCoA weighted UniFrac distances
plot_PCoA_wu_microaerob <- plot_ordination(physeq = psdata_RBRA_rel_microaerob, 
                                 ordination = PCoA_wu_microaerob, 
                                 type = "samples", 
                                 axes = c(1,2), 
                                 shape = "day_doc",
                                 color = "column_height") +
  scale_color_manual(name = "Column_height",
                     values = c("#CA0020")) +
  scale_shape_manual(name = "Day_DOC",
                       values = c(0,1,5,4)) +
  geom_point(size = 2) +
  ggtitle("weighted UniFrac") +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))
```

#### Distances between consecutive electron donors

```{r distances electron donors, message=F, echo=T, eval=T, warning=F, include=T, cache=F}
# subtract sample metadata from phyloseq object
metadata2 <- data.frame(sample_data(psdata_RBRA_rel_donor), stringsAsFactors = T)

# calculate distance matrices
dist_bc = vegdist(t(otu_table(psdata_RBRA_rel_donor)), method = "bray", binary = T)
dist_jac = vegdist(t(otu_table(psdata_RBRA_rel_donor)), method = "jaccard", binary = T)
dist_uu = phyloseq::UniFrac(psdata_RBRA_rel_donor, weighted = F)
dist_wu = phyloseq::UniFrac(psdata_RBRA_rel_donor, weighted = T)

# reshape to long format
bc_long =
dist_bc %>%
  as.matrix() %>%
  as_tibble(rownames = "sample_a") %>%
  pivot_longer(-sample_a, names_to = "sample_b", values_to = "distance") %>%
  mutate(metric = "bray")

jaccard_long =
dist_jac %>%
  as.matrix() %>%
  as_tibble(rownames = "sample_a") %>%
  pivot_longer(-sample_a, names_to = "sample_b", values_to = "distance") %>%
  mutate(metric = "jaccard")

uu_long =
dist_uu %>%
  as.matrix() %>%
  as_tibble(rownames = "sample_a") %>%
  pivot_longer(-sample_a, names_to = "sample_b", values_to = "distance") %>%
  mutate(metric = "uu")

wu_long =
dist_wu %>%
  as.matrix() %>%
  as_tibble(rownames = "sample_a") %>%
  pivot_longer(-sample_a, names_to = "sample_b", values_to = "distance") %>%
  mutate(metric = "wu")

# define consecutive sampling moments
comp_donor = c("H_14_to_H_D_42", "H_D_42_to_D_70", "D_70_to_D_A_98", "D_A_98_to_A_126", "A_126_to_NH4_154")

# reshape distance data and combine with metadata
all_dists =
bind_rows(wu_long, uu_long, jaccard_long, bc_long) %>%
  filter(sample_a != sample_b) %>% # remove self-comparisons
  inner_join(., metadata2 %>% dplyr::select(sampleid, day_doc, column_height), by = c("sample_a" = "sampleid")) %>%
  rename(day_doc_a = day_doc,
         column_height_a = column_height) %>% 
   inner_join(., metadata2 %>% dplyr::select(sampleid, day_doc, column_height), by = c("sample_b" = "sampleid")) %>%
  rename(day_doc_b = day_doc,
         column_height_b = column_height) %>% 
  mutate(comp = paste0(day_doc_a, "_to_", day_doc_b)) %>%
  filter(comp %in% comp_donor) %>% 
  mutate(comp = factor(comp, levels = comp_donor)) %>%
  filter(column_height_a == column_height_b)

# plot Bray-Curtis distances between consecutive e-donors
comp_inoc_bc <- all_dists %>%
  filter(metric == "bray") %>% 
  ggplot(aes(x = comp, y = distance, color = column_height_a)) +
  geom_point(size = 2, alpha = 0.6, position = position_dodge(0.05)) +
  ggtitle("Bray Curtis distances") +
  scale_color_manual(values = c("#F4A582", "#92C5DE", "#0571B0")) +
  scale_fill_manual(values = c("#F4A582", "#92C5DE", "#0571B0")) +
  scale_y_continuous(limits = c(0, 0.75), 
                     breaks = seq(0, 0.75, 0.25)) +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(5, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5),
        legend.position = "right") +
  labs(y = "Distance",
       x = NULL)

# plot Jaccard distances between consecutive e-donors
comp_inoc_jac <- all_dists %>%
  filter(metric == "jaccard") %>% 
  ggplot(aes(x = comp, y = distance, color = column_height_a)) +
  geom_point(size = 2, alpha = 0.6, position = position_dodge(0.05)) +
  ggtitle("Jaccard distances") +
  scale_color_manual(values = c("#F4A582", "#92C5DE", "#0571B0")) +
  scale_fill_manual(values = c("#F4A582", "#92C5DE", "#0571B0")) +
  scale_y_continuous(limits = c(0, 0.75), 
                     breaks = seq(0, 0.75, 0.25)) +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(5, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5),
        legend.position = "right") +
  labs(y = "Distance",
       x = NULL)

# plot unweighted UniFrac distances between consecutive e-donors
comp_inoc_uu <- all_dists %>%
  filter(metric == "uu") %>%  
  ggplot(aes(x = comp, y = distance, color = column_height_a)) +
  geom_point(size = 2, alpha = 0.6, position = position_dodge(0.05)) +
  ggtitle("unweighted UniFrac distances") +
  scale_color_manual(values = c("#F4A582", "#92C5DE", "#0571B0")) +
  scale_fill_manual(values = c("#F4A582", "#92C5DE", "#0571B0")) +
  scale_y_continuous(limits = c(0, 0.75), 
                     breaks = seq(0, 0.75, 0.25)) +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(5, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5),
        legend.position = "right") +
  labs(y = "Distance",
       x = NULL)

# plot weighted UniFrac distances between consecutive e-donors
comp_inoc_wu <- all_dists %>%
  filter(metric == "wu") %>%  
  ggplot(aes(x = comp, y = distance, color = column_height_a)) +
  geom_point(size = 2, alpha = 0.6, position = position_dodge(0.05)) +
  ggtitle("weighted UniFrac distances") +
  scale_color_manual(values = c("#F4A582", "#92C5DE", "#0571B0")) +
  scale_fill_manual(values = c("#F4A582", "#92C5DE", "#0571B0")) +
  scale_y_continuous(limits = c(0, 0.75), 
                     breaks = seq(0, 0.75, 0.25)) +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(5, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5),
        legend.position = "right") +
  labs(y = "Distance",
       x = NULL)
```

#### Distances between consecutive microaeration periods

```{r distances microaeration periods, message=F, echo=T, eval=T, warning=F, include=T, cache=F}
# subtract sample metadata from phyloseq object
metadata3 <- data.frame(sample_data(psdata_RBRA_rel_microaerob), stringsAsFactors = T)

# calculate distance matrices
dist_bc2 = vegdist(t(otu_table(psdata_RBRA_rel_microaerob)), method = "bray", binary = T)
dist_jac2 = vegdist(t(otu_table(psdata_RBRA_rel_microaerob)), method = "jaccard", binary = T)
dist_uu2 = phyloseq::UniFrac(psdata_RBRA_rel_microaerob, weighted = F)
dist_wu2 = phyloseq::UniFrac(psdata_RBRA_rel_microaerob, weighted = T)

# reshape to long format
bc_long2 =
dist_bc2 %>%
  as.matrix() %>%
  as_tibble(rownames = "sample_a") %>%
  pivot_longer(-sample_a, names_to = "sample_b", values_to = "distance") %>%
  mutate(metric = "bray")

jaccard_long2 =
dist_jac2 %>%
  as.matrix() %>%
  as_tibble(rownames = "sample_a") %>%
  pivot_longer(-sample_a, names_to = "sample_b", values_to = "distance") %>%
  mutate(metric = "jaccard")

uu_long2 =
dist_uu2 %>%
  as.matrix() %>%
  as_tibble(rownames = "sample_a") %>%
  pivot_longer(-sample_a, names_to = "sample_b", values_to = "distance") %>%
  mutate(metric = "uu")

wu_long2 =
dist_wu2 %>%
  as.matrix() %>%
  as_tibble(rownames = "sample_a") %>%
  pivot_longer(-sample_a, names_to = "sample_b", values_to = "distance") %>%
  mutate(metric = "wu")

# define consecutive sampling moments
comp_microaerob = c("Nitrate_red_14_to_Microaerob_70", "Microaerob_70_to_Microaerob_126", "Microaerob_126_to_Microaerob_154")

# reshape distance data and combine with metadata
all_dists2 =
bind_rows(wu_long2, uu_long2, jaccard_long2, bc_long2) %>%
  filter(sample_a != sample_b) %>% # remove self-comparisons
  inner_join(., metadata3 %>% dplyr::select(sampleid, day_doc, column_height), by = c("sample_a" = "sampleid")) %>%
  rename(day_doc_a = day_doc,
         column_height_a = column_height) %>% 
   inner_join(., metadata3 %>% dplyr::select(sampleid, day_doc, column_height), by = c("sample_b" = "sampleid")) %>%
  rename(day_doc_b = day_doc,
         column_height_b = column_height) %>% 
  mutate(comp = paste0(day_doc_a, "_to_", day_doc_b)) %>%
  filter(comp %in% comp_microaerob) %>% 
  mutate(comp = factor(comp, levels = comp_microaerob)) %>%
  filter(column_height_a == column_height_b)

# plot Bray-Curtis distances between consecutive e-donors
comp_inoc_bc2 <- all_dists2 %>%
  filter(metric == "bray") %>% 
  ggplot(aes(x = comp, y = distance, color = column_height_a)) +
  geom_point(size = 2, alpha = 0.6, position = position_dodge(0.05)) +
  ggtitle("Bray Curtis distances") +
  scale_color_manual(values = c("#CA0020")) +
  scale_fill_manual(values = c("#CA0020")) +
  scale_y_continuous(limits = c(0, 1), 
                     breaks = seq(0, 1, 0.25)) +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(5, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5),
        legend.position = "right") +
  labs(y = "Distance",
       x = NULL)

# plot Jaccard distances between consecutive e-donors
comp_inoc_jac2 <- all_dists2 %>%
  filter(metric == "jaccard") %>% 
  ggplot(aes(x = comp, y = distance, color = column_height_a)) +
  geom_point(size = 2, alpha = 0.6, position = position_dodge(0.05)) +
  ggtitle("Jaccard distances") +
  scale_color_manual(values = c("#CA0020")) +
  scale_fill_manual(values = c("#CA0020")) +
  scale_y_continuous(limits = c(0, 1), 
                     breaks = seq(0, 1, 0.25)) +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(5, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5),
        legend.position = "right") +
  labs(y = "Distance",
       x = NULL)

# plot unweighted UniFrac distances between consecutive e-donors
comp_inoc_uu2 <- all_dists2 %>%
  filter(metric == "uu") %>%  
  ggplot(aes(x = comp, y = distance, color = column_height_a)) +
  geom_point(size = 2, alpha = 0.6, position = position_dodge(0.05)) +
  ggtitle("unweighted UniFrac distances") +
  scale_color_manual(values = c("#CA0020")) +
  scale_fill_manual(values = c("#CA0020")) +
  scale_y_continuous(limits = c(0, 1), 
                     breaks = seq(0, 1, 0.25)) +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(5, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5),
        legend.position = "right") +
  labs(y = "Distance",
       x = NULL)

# plot weighted UniFrac distances between consecutive e-donors
comp_inoc_wu2 <- all_dists2 %>%
  filter(metric == "wu") %>%  
  ggplot(aes(x = comp, y = distance, color = column_height_a)) +
  geom_point(size = 2, alpha = 0.6, position = position_dodge(0.05)) +
  ggtitle("weighted UniFrac distances") +
  scale_color_manual(values = c("#CA0020")) +
  scale_fill_manual(values = c("#CA0020")) +
  scale_y_continuous(limits = c(0, 1), 
                     breaks = seq(0, 1, 0.25)) +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(5, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5),
        legend.position = "right") +
  labs(y = "Distance",
       x = NULL)
```

### Figure 3 - Alpha + Beta diversity

```{r Figure 3, message=F, echo=T, eval=T, warning=F, include=T, cache=F, fig.width=11, fig.height=7}
legend_beta_donor <- get_legend(plot_PCoA_Jac_donor)

legend_beta_microaerob <- get_legend(plot_PCoA_Jac_microaerob)

legend_comp <- get_legend(comp_inoc_jac)

legend_comp2 <- get_legend(comp_inoc_jac2)

PCoA <- plot_grid(plot_PCoA_wu_microaerob + theme(legend.position = "none"),
                 legend_beta_microaerob,
                 plot_PCoA_Jac_donor + theme(legend.position = "none"),
                 plot_PCoA_uu_donor + theme(legend.position = "none"),
                 plot_PCoA_wu_donor + theme(legend.position = "none"),
                 legend_beta_donor,
                 comp_inoc_wu2 + theme(legend.position = "none"),
                 legend_comp2,
                 comp_inoc_jac + theme(legend.position = "none"),
                 comp_inoc_uu + theme(legend.position = "none"),
                 comp_inoc_wu + theme(legend.position = "none"),
                 legend_comp,
                 labels  = c("C", "", "D", "E", "F", "", "", "", "", "", "", ""),
                 ncol = 6,
                 nrow = 2,
                 align = "hv",
                 rel_widths = c(1,0.5,1,1,1,0.5),
                 rel_heights = c(1,0.95))

fig_3 <- plot_grid (alpha_plot,
                    PCoA,
                    nrow = 2,
                    align = "hv",
                    rel_heights = c(1,1.5))

fig_3

ggsave("C:/Users/Rita/Documents/R/amendment_micropollutant_biodegradation_aquifer/figures/Figure4_3.pdf", width = 11, height = 7)
```

**Figure 3**. Microbial community diversity across the different columns amendments. Alpha diversity expressed as A) ASV richness (Chao1) and B) ASV diversity (Shannon). Principal coordinates analysis (PCoA) ordination plots of community composition and corresponding distances between microbial communities selected under consecutive redox conditions or by consecutive electron donor amendments based on C) weighted UniFrac distances for the microaerobic cycles without electron donor amendment column; and based on D) Jaccard; E) unweighted UniFrac; and F) weighted UniFrac distances, respectively, for the remaining columns. Distance denotes the pairwise difference in community composition (i.e. beta diversity), where 0 and 1 represent identical and entirely different community compositions, respectively.<br>

### Figure S6 - Remaining Beta diversity

```{r Figure S6, message=F, echo=T, eval=T, warning=F, include=T, cache=F, fig.width=11, fig.height=4.2}
PCoA_sm <- plot_grid(plot_PCoA_Jac_microaerob + theme(legend.position = "none"),
                   plot_PCoA_BC_microaerob + theme(legend.position = "none"),
                   plot_PCoA_uu_microaerob + theme(legend.position = "none"),
                   legend_beta_microaerob,
                   plot_PCoA_BC_donor + theme(legend.position = "none"),
                   legend_beta_donor,
                   comp_inoc_jac2 + theme(legend.position = "none"),
                   comp_inoc_bc2 + theme(legend.position = "none"),
                   comp_inoc_uu2 + theme(legend.position = "none"),
                   legend_comp2,
                   comp_inoc_bc + theme(legend.position = "none"),
                   legend_comp,
                   labels  = c("A", "B", "C", "", "D", "", "", "", "", "", "", ""),
                   ncol = 6,
                   nrow = 2,
                   align = "hv",
                   rel_widths = c(1,1,1,0.5,1,0.5),
                   rel_heights = c(1,0.95))
PCoA_sm

ggsave("C:/Users/Rita/Documents/R/amendment_micropollutant_biodegradation_aquifer/figures/Figure4_S6.pdf", width = 11, height = 4.2)
```

**Figure S6**. Microbial community diversity across the different column amendments. Principal coordinates analysis (PCoA) ordination plots of community composition and corresponding distances between microbial communities selected under consecutive redox conditions or by consecutive electron donor amendments based on A) Jaccard; B) Bray Curtis; and C) unweighted UniFrac distances, respectively, for the microaerobic cycles without electron donor amendment column; and based on D) Bray Curtis distances for the remaining columns. Distance denotes the pairwise difference in community composition (i.e. beta diversity), where 0 and 1 represent identical and entirely different community compositions, respectively.<br>

### Relative abundance

```{r rel abund barplot class, message=F, echo=T, eval=T, warning=T, include=T, cache=F, fig.width=8, fig.height=4}
# input data
ps_Class <- psmelt(transform_sample_counts(tax_glom(psdata_RBRA, "Class"), fun = function(x) x/sum(x)))

# relative abundance based on Class
class_count <- ps_Class %>% 
  as_tibble() %>% 
  mutate(Class = factor(Class)) %>% 
  pull(Class) %>% 
  levels() # count class n = 80

Class_abundances <- ps_Class %>% 
  as_tibble() %>% 
  dplyr::select(sampleid, day_doc, column_height, Class, Abundance) %>% 
  mutate(day_doc = as.character(day_doc),
         column_height = as.character(column_height)) %>% 
  group_by(sampleid, day_doc, column_height, Class) %>% 
  summarize(rel_abund = sum(Abundance), .groups = "drop") %>% 
  group_by(day_doc,column_height, Class) %>% 
  summarize(mean_rel_abund = 100* mean(rel_abund), .groups = "drop") %>% 
  mutate(Class = str_replace(Class, "(.*)_unClassified", "UnClassified *\\1*"),
         Class = str_replace(Class, "^(\\S*)$", "*\\1*"))

Class_pool <- Class_abundances %>% 
  group_by(Class) %>% 
  summarize(pool = max(mean_rel_abund) < 4, 
            mean = mean(mean_rel_abund), 
            .groups = "drop")

# plot relative abundance based on Class
plot_Class <- inner_join(Class_abundances, Class_pool, by="Class") %>% 
  mutate(Class = if_else(pool, "Other", Class)) %>% 
  group_by(day_doc, column_height, Class) %>%
  summarize(mean_rel_abund = sum(mean_rel_abund), 
            mean = min(mean),
            .groups = "drop") %>% 
  mutate(Class = factor(Class), 
         Class = fct_reorder(Class, mean, .desc = T),
         day_doc = factor(day_doc),
         day_doc = fct_relevel(day_doc, rev(c("14_H", "42_H_D",	"70_D",	"98_D_A",	"126_A",	"154_NH4",	"14_Nitrate_red",	"70_Microaerob", "126_Microaerob",	"154_Microaerob"))),
         column_height = factor(column_height),
         column_height = fct_relevel(column_height, rev(c("Nitrate_red_Microaer_50",	"Nitrate_red_2.5", "Aerob_2.5", "Aerob_Microaer_2.5", "Aerob_Microaer_50")))) %>% 
ggplot(aes(x=day_doc, y = mean_rel_abund, fill = Class)) +
  geom_col() +
  scale_fill_manual(name=NULL, 
                    values = c("#B2182B", "#D6604D", "#2166AC", "#4393C3", "#4D4D4D", "#878787", "#01665E", "#35978F", "#ABDDA4", "#E6F598", "#8C510A", "#BF812D", "#C51B7D", "#DE77AE", "#4D9221", "#7FBC41",  "#762A83", "#9970AB", "#FDAE61", "#FEE090",  "#1B7837", "#5AAE61")) +
  facet_wrap(~column_height, ncol = 2, scales = "free") +
  scale_y_continuous(expand = c(0,0)) +
  labs(x="Day_DOC", y="Mean Relative Abundance (%)") +
  theme_classic() +
  theme(axis.text = element_text(size = 7, colour = "black"),
        axis.ticks.x = element_line(size = 0.35),
        axis.ticks.y = element_line(size = 0.35),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text = element_markdown(size=7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5)) +
  coord_flip()

class = plot_grid(plot_Class + theme(legend.position = "none"), 
                   ncol = 1, 
                   nrow = 1, 
                   align = "hv")

legend_class <- get_legend(plot_Class + guides(fill = guide_legend(ncol = 8)) +
    theme(legend.position = "right"))

plot_grid(class, legend_class, nrow=2, rel_heights = c(5, 1))

ggsave("C:/Users/Rita/Documents/R/amendment_micropollutant_biodegradation_aquifer/figures/Figure4_abundance_class.pdf", width = 8, height = 4)
```

**Figure abundance class**. Microbial community composition based on Class occurrences  X most abundant class.<br>

```{r rel abund barplot genus, message=F, echo=T, eval=T, warning=T, include=T, cache=F, fig.width=10, fig.height=4}
# input data
ps_Genus <- psmelt(transform_sample_counts(tax_glom(psdata_RBRA, "Genus"), fun = function(x) x/sum(x)))

# relative abundance based on Genus
Genus_count <- ps_Genus %>% 
  as_tibble() %>% 
  mutate(Genus = factor(Genus)) %>% 
  pull(Genus) %>% 
  levels() # count Genus n = 80

Genus_abundances <- ps_Genus %>% 
  as_tibble() %>% 
  dplyr::select(sampleid, day_doc, column_height, Genus, Abundance) %>% 
  mutate(day_doc = as.character(day_doc),
         column_height = as.character(column_height)) %>% 
  group_by(sampleid, day_doc, column_height, Genus) %>% 
  summarize(rel_abund = sum(Abundance), .groups = "drop") %>% 
  group_by(day_doc,column_height, Genus) %>% 
  summarize(mean_rel_abund = 100* mean(rel_abund), .groups = "drop") %>% 
  mutate(Genus = str_replace(Genus, "(.*)_unClassified", "UnClassified *\\1*"),
         Genus = str_replace(Genus, "^(\\S*)$", "*\\1*"))

Genus_pool <- Genus_abundances %>% 
  group_by(Genus) %>% 
  summarize(pool = max(mean_rel_abund) < 10, 
            mean = mean(mean_rel_abund), 
            .groups = "drop")

# plot relative abundance based on Genus
plot_Genus <- inner_join(Genus_abundances, Genus_pool, by="Genus") %>% 
  mutate(Genus = if_else(pool, "Other", Genus)) %>% 
  group_by(day_doc, column_height, Genus) %>%
  summarize(mean_rel_abund = sum(mean_rel_abund), 
            mean = min(mean),
            .groups = "drop") %>% 
  mutate(Genus = factor(Genus), 
         Genus = fct_reorder(Genus, mean, .desc = T),
         day_doc = factor(day_doc),
         day_doc = fct_relevel(day_doc, rev(c("14_H", "42_H_D",	"70_D",	"98_D_A",	"126_A",	"154_NH4",	"14_Nitrate_red",	"70_Microaerob", "126_Microaerob",	"154_Microaerob"))),
         column_height = factor(column_height),
         column_height = fct_relevel(column_height, rev(c("Nitrate_red_Microaer_50",	"Nitrate_red_2.5", "Aerob_2.5", "Aerob_Microaer_2.5", "Aerob_Microaer_50")))) %>% 
ggplot(aes(x=day_doc, y = mean_rel_abund, fill = Genus)) +
  geom_col() +
  scale_fill_manual(name=NULL, 
                    values = c("#B2182B", "#D6604D", "#2166AC", "#4393C3", "#4D4D4D", "#878787", "#01665E", "#35978F", "#ABDDA4", "#E6F598", "#8C510A", "#BF812D", "#C51B7D", "#DE77AE", "#4D9221", "#7FBC41",  "#762A83", "#9970AB", "#FDAE61", "#FEE090",  "#1B7837", "#5AAE61")) +
  facet_wrap(~column_height, ncol = 2, scales = "free") +
  scale_y_continuous(expand = c(0,0)) +
  labs(x="Day_DOC", y="Mean Relative Abundance (%)") +
  theme_classic() +
  theme(axis.text = element_text(size = 7, colour = "black"),
        axis.ticks.x = element_line(size = 0.35),
        axis.ticks.y = element_line(size = 0.35),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text = element_markdown(size=7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5)) +
  coord_flip()

Genus = plot_grid(plot_Genus + theme(legend.position = "none"), 
                   ncol = 1, 
                   nrow = 1, 
                   align = "hv")

legend_Genus <- get_legend(plot_Genus + guides(fill = guide_legend(ncol = 8)) +
    theme(legend.position = "right"))

plot_grid(Genus, legend_Genus, nrow=2, rel_heights = c(5, 1))

ggsave("C:/Users/Rita/Documents/R/amendment_micropollutant_biodegradation_aquifer/figures/Figure4_abundance_genus.pdf", width = 8, height = 4)
```

**Figure abundance genus**. Microbial community composition based on Genus occurrences  X most abundant genus.<br>

### Capscale analysis

Capscale was performed with partialling out x, i.e. focused on y

```{r partial analysis, message=F, echo=T, eval=T, warning=T, include=F, cache=F}
# load phyloseq object
phy_rita = psdata_RBRA

# chemical data
chem_data = readxl::read_xlsx("C:/Users/Rita/Documents/R/amendment_micropollutant_biodegradation_aquifer/input_data/Manuscript_4_micropol_edonor.xlsx")

# sample_data
sam_dat = 
  sample_data(phy_rita) %>% data.frame() %>% dplyr::select(SampleID = sampleid, everything()) %>% tibble

# joined data
chem_data_added = sam_dat %>% inner_join(., chem_data, by = "SampleID")

# overwriting old metadata
sam_data = sample_data(chem_data_added)
sample_names(sam_data) = chem_data_added$SampleID
sample_data(phy_rita) = sam_data

# rarefy data
min_ss = min(sample_sums(phy_rita))

set.seed(711)
phy_rita_32198 = rarefy_even_depth(phy_rita, sample.size = min_ss, rngseed = 711, trimOTUs = T)

# total number of reads
tot_read = phy_rita_32198 %>% sample_sums(.) %>% sum()

# create taxa acronyms
df_taxa_names = data.frame(OTU = taxa_names(phy_rita_32198),
                           ASV = paste0("ASV_",seq(1:ntaxa(phy_rita_32198))))

# rename colnames with acronyms
abund = t(otu_table(phy_rita_32198))
cap_data = abund/rowSums(abund)
colnames(cap_data) = df_taxa_names$ASV

rownames(chem_data_added) = chem_data_added$SampleID #Warning: Setting row names on a tibble is deprecated.

# predictors
chem_pred = 
  chem_data_added %>% 
  dplyr::select(column_height, day_doc, `24_D`:MET_OA) %>% 
  mutate(across(where(is.numeric), ~if_else(.<0, 0, .))) %>% 
  mutate(across(where(is.numeric), ~scale(.)),
         day_doc = factor(day_doc, levels = c("14_H", "42_H_D",	"70_D",	"98_D_A",	"126_A",	"154_NH4",	"14_Nitrate_red",	"70_Microaerob", "126_Microaerob",	"154_Microaerob")))

#model definitions (all predictor removal efficiencies are added)
full.model <- capscale(cap_data ~
                         `24_D` + MCPP,
                       chem_pred[,-c(1:2)],
                       dist="jaccard",
                       na.action = na.exclude)

doc.model <- capscale(cap_data ~ 
                         `24_D` + MCPP +
                          Condition(column_height), # partial analysis now
                        chem_pred[,-2], 
                        dist="jaccard",
                        na.action = na.exclude)

column_height.model <- capscale(cap_data ~
                  `24_D` + MCPP +
                  Condition(day_doc), # partial analysis now
                chem_pred[,-1],
                dist="jaccard",
                na.action = na.exclude)

vif_cca <- vif.cca(doc.model)
```

```{r model selection, message=F, echo=T, eval=T, warning=T, include=F, cache=F}
# model selection
selectedMod_full <- step(full.model)
selectedMod_ch <- step(column_height.model)
selectedMod_doc <- step(doc.model)
smry = summary(doc.model)
```

```{r ANOVA table, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# general ANOVA
anova_tab_gen = anova(selectedMod_doc) %>% as_tibble(rownames = "terms")

# e-donor ANOVA
anova_tab_doc = anova(selectedMod_doc, by = "terms") %>% as_tibble(rownames = "terms")
anova_tab_doc = anova_tab_doc %>% 
  mutate(R2 = 100*(SumOfSqs/sum(SumOfSqs))) %>% 
  mutate(fdr_q = p.adjust(`Pr(>F)`, "fdr")) %>% 
  filter(fdr_q < 0.1 | is.na(fdr_q)) # correct p values for multiple testing filter(fdr_q < 0.1)

anova_tab_doc

anova_tab_doc_kable <- anova_tab_doc %>% 
  kableExtra::kbl(digits = 3, 
                  caption = "Constrained ordination using db-RDA (Jaccard distances)<br> Associations between microbial community membership and MP removal efficiencies<br>Terms = MPs with FDR q<0.1 (stepwise selection)") %>% 
  kableExtra::kable_classic()

anova_tab_doc_kable
```

**Table S7**.Constrained ordination using db-RDA (Jaccard distances). Associations between microbial community membership and MP removal efficiencies. Terms = MPs with FDR q<0.1 (stepwise selection).<br>

```{r capscale analysis - CAP plot, message=F, echo=T, eval=T, warning=T, include=T, cache=F, fig.width=12, fig.height=6}
# plot dbRDA model focused on redox_conditions
plot_data = chem_data_added %>% dplyr::select(SampleID, Sample_type,Redox_type, any_of(smry$biplot %>% rownames()))
df1  <- data.frame(smry$sites[,1:2]) %>% as_tibble(rownames = "SampleID") # CAP1 and CAP2
plot_data = df1 %>% inner_join(., plot_data, by = "SampleID" )

## plot CAP ordinations
### axis 1 vs 2
ord_red_ax12 = 
  ggord(redox.model,
        xlims = c(-1.1, 0.5),
        ylims = c(-0.8, 0.6),
        arrow = 0.2,
        axes = c("1","2"),
        grp_in = plot_data$Redox_type,
        ellipse =F, 
        coord_fix = T,
        addsize = 0,
        add_col = NULL,
        ext = 1.2,
        vec_ext = 0.8,
        var_sub = anova_tab_red$terms[-length(anova_tab_red$terms)]) +
  labs(title = "Partial dbRDA - effect of Redox condition; CAP1 vs CAP2")

### axis 1 vs 3
ord_red_ax13 = 
  ggord(redox.model,
        xlims = c(-1.1, 0.5),
        ylims = c(-0.9, 0.6),
        arrow = 0.2,
        axes = c("1","3"),
        grp_in = plot_data$Redox_type,
        ellipse =F, 
        coord_fix = T,
        addsize = 0,
        ext = 1.2,
        vec_ext = 0.8,
        var_sub = anova_tab_red$terms[-length(anova_tab_red$terms)]) +
  labs(title = "Partial dbRDA - effect of Redox condition; CAP1 vs CAP3")

### make ggord, remove points layer and add shapes etc
ord_red_ax12$layers[[1]] <- NULL # to adjust shape as well
ord_red_ax13$layers[[1]] <- NULL # to adjust shape as well

ord_red_ax12$data <- ord_red_ax12$data %>% 
  mutate(newgrp = plot_data$Sample_type, # add extra group for sample_type
         samplecheck = plot_data$SampleID,
         Groups = factor(Groups, levels = c("aerobic", "nitrate", "iron", "sulphate", "methanogenic")),
         newgrp = factor(newgrp, levels = c("soil", "mun", "ditch", "ind")))

ord_red_ax13$data <- ord_red_ax13$data %>% 
  mutate(newgrp = plot_data$Sample_type, # add extra group for sample_type
         samplecheck = plot_data$SampleID,
         Groups = factor(Groups, levels = c("aerobic", "nitrate", "iron", "sulphate", "methanogenic")),
         newgrp = factor(newgrp, levels = c("soil", "mun", "ditch", "ind")))

# plot CAP ordinations again
shapes_red = c(22, 21, 4, 24, 25)
colors_red = c("#C92127", "#F3A484", "#93C4DD", "#0772B2")

plot_ord_red_ax12 = 
  ord_red_ax12 + geom_point(aes(shape = Groups, colour = newgrp, group = newgrp), size=2, stroke=1.5, alpha=1) + 
  scale_shape_manual(name = "",
                     values = shapes_red) +
  scale_color_manual(name = "",
                     values = colors_red,
                     labels = c("Soil","Mun AS","Ditch", "Ind AS")) + 
  guides(fill = guide_legend(override.aes= list(shape = NA))) + 
  theme(legend.title = element_blank())

plot_ord_red_ax13 = 
  ord_red_ax13 + geom_point(aes(shape = Groups, colour = newgrp, group = newgrp), size=2, stroke=1.5, alpha=1) + 
  scale_shape_manual(name = "",
                     values = shapes_red) +
  scale_color_manual(name = "",
                     values = colors_red,
                     labels = c("Soil","Mun AS","Ditch", "Ind AS")) + 
  guides(fill = guide_legend(override.aes= list(shape = NA))) + 
  theme(legend.title = element_blank())

legend_bar_CAP = 
  get_legend(plot_ord_red_ax12 + theme( legend.position = "bottom",
                                        legend.box = "vertical",
                                        legend.margin=margin(),
                                        legend.key.size = unit(1, 'cm'),
                                        legend.key.height = unit(1, 'cm'),
                                        legend.key.width = unit(1, 'cm'),
                                        legend.title = element_text(size=0),
                                        legend.text = element_text(size=12)))

CAP_plot = 
  plot_grid(
    plot_ord_red_ax12 + theme(legend.position = "none"), 
    plot_ord_red_ax13 + theme(legend.position = "none"), 
    labels  = c("A", "B"), 
    ncol = 2, 
    nrow = 1, 
    align = "hv")


plot_grid(CAP_plot, legend_bar_CAP, ncol=1, nrow = 2, rel_heights = c(5, 1))

ggsave("C:/Users/Rita/Documents/R/amendment_micropollutant_biodegradation_aquifer/figures/Figure4_cap.pdf", width = 12, height = 5)
```

**Figure CAP**. Redundancy analysis linking micropollutant removal efficiency to microbial community composition. A) CAP1 vs CAP2 and B) CAP1 vs CAP3.<br>